src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -91,9 +128,8 @@
       <NetSdkSupportingFeed>https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet9/nuget/v3/index.json</NetSdkSupportingFeed>
     </PropertyGroup>
 
-    <!-- Update the detected or manually specified NuGetConfigFile, but also allow multiple. -->
     <ItemGroup>
-      <NuGetConfigFiles Include="$(NuGetConfigFile)" />
+      <EnvironmentVariables Include="RestoreConfigFile=$(RestoreConfigFile)" />
```

Comment:
Now with the work that you put into making repos flow the property to inner Exec tasks, I think you should remove the env var here.

Reply:
We can't remove it due to various tools projects in `aspnetcore`, `fsharp` and `source-build-externals`. With additional effort it might be possible to eventually get rid of the env.var. I'd prefer to have that as a follow up work-item - https://github.com/dotnet/source-build/issues/4081